### DISCUSSION TITLE
Another Extraordinarily basic Question (regarding Get output)

### DISCUSSION THREAD

**Discussion by Michael-132-wq** [bonsai-rx/bonsai#2040] [Q&A] [answered]
Hello, 

How do I set a pin on an Arduino based on a flag that is set in a python script. This seems incredibly straight forward 

Basic documentation would be hugely helpful 

![Screenshot 2024-10-17 105230](https://github.com/user-attachments/assets/d9ae3ae9-2d9a-4481-9c4c-f3f5a272ce5b)

**Comment by bruno-f-cruz** [comment]
Hi!

I can't access your code since you only posted an image. As suggested in #864, can you please include the code as well? 

Looking at the image, I believe that the only thing you need to ensure at the output of "Get" is that its type is a Boolean. `DigitalOutput` takes boolean as input.

You can see examples of this operator here: https://bonsai-rx.org/docs/tutorials/closed-loop.html#exercise-2-measuring-video-acquisition-latency
and the source code here: https://github.com/bonsai-rx/arduino/blob/c5f605a43247d2fea02f43a3cf7e5b6dd3b71d20/Bonsai.Arduino/DigitalOutput.cs#L45C23-L46C63

Finally, would you mind changing the title of the post to something more descriptive as documented in #864 

Thanks!

**Reply by Michael-132-wq** [reply]
That is exactly what I need to do. How do you convert the get operator output to bool? 

The variable in python is a bool

but get outputs "python runtime object"

**Reply by rachelmcassidy** [reply]
I have the same question!

**Reply by ncguilbeault** [reply, marked as answer]
Currently, the way to do this is to explicitly convert the `PyObject` into the corresponding .NET data type using something like a `ExpressionTransform` node. For example:

![image](https://github.com/user-attachments/assets/df19b4e2-8146-43d8-ace3-06acf30d9901)

Inside the `ExpressionTransform`, you would write something like:

```
Convert.ToBoolean(it)
```

or if it is a number (i.e. double, float, etc):

```
Convert.ToDouble(it)
```

This method works well for simple data types, but if you need to convert a complex data type, your best bet is to write a conversion function in C# (using a `CSharpTransform` node or writing a custom package). As an example, for the Bonsai.ML package,  there is the `ConvertPythonObjectToCSharp` function which can convert different `PyObject` types into .NET types, including NumPy arrays: https://github.com/bonsai-rx/machinelearning/blob/886150001b7635d804de54af0dad1bb5a79bacc9/src/Bonsai.ML.Python/PythonHelper.cs#L45
